File: internal/content/testdata/bench/typescript/toggle-case.ts
function toggle_case(s: string): string {
    let result: string[] = [];
    for (const ch of s) {
        if (ch === ch.toUpperCase() && ch !== ch.toLowerCase()) {
            result.push(ch.toLowerCase());
        } else if (ch === ch.toLowerCase() && ch !== ch.toUpperCase()) {
            result.push(ch.toUpperCase());
        } else {
            result.push(ch);
        }
    }
    return result.join('');
}

File: internal/content/testdata/bench/go/toggle-case.go
func toggle_case(s string) string {
	result := make([]byte, len(s))
	for i := 0; i < len(s); i++ {
		ch := s[i]
		if ch >= 'A' && ch <= 'Z' {
			result[i] = ch + 32
		} else if ch >= 'a' && ch <= 'z' {
			result[i] = ch - 32
		} else {
			result[i] = ch
		}
	}
	return string(result)
}
